src/fst.jl

```diff
@@ -1120,14 +1120,14 @@ for i = 1:10 body end
 
 - https://github.com/domluna/JuliaFormatter.jl/issues/34
 """
-function eq_to_in_normalization!(fst::FST, always_for_in::Bool)
+function eq_to_in_normalization!(fst::FST, always_for_in::Bool, for_in_replacement::String)
     if fst.typ === Binary
         idx = findfirst(n -> n.typ === OPERATOR, fst.nodes)
         idx === nothing && return
         op = fst[idx]
 
-        if always_for_in
-            op.val = "in"
+        if always_for_in && valid_for_in_op(op.val)
```

Comment:
the validation should occur when the options are initially set. And if the op is invalid it should display a message and not format.

Reply:
Hm, this is taken from:
https://github.com/domluna/JuliaFormatter.jl/pull/380/files#diff-fdbcedac9cc0bafcd2fa355f83f9558522f56e8e432e9a3775f59612dce9fe8aR1016

I assumed that it validates the "incoming" operator (that we're not rewriting something that's not a cycle), but actually that doesn't make sense because that wouldn't ever get parsed. This condition doesn't make sense here then, will change shortly.

Reply:
OK this is gonna be harder, the validation was there because of this:
```
always convert `=` to `in` (for loops): Test Failed at /home/exa/work/JuliaFormatter.jl/test/options.jl:253
  Expression: fmt(str_, always_for_in = true) == str
   Evaluated: "[i for i in 1:10 if i in 2]" == "[i for i in 1:10 if i == 2]"
```
(i.e., it force-replaces the == operator even in the `if` part)

Any best way to dodge this? (I pushed the failing version now)

Reply:
you shouldn't need this check the `valid_for_in_op(op.val)` part of `if always_for_in && valid_for_in_op(op.val)` anymore since it's being checked when the options are created.

Reply:
the `valid_for_in_op` check in the condition is gone (see the new commits); the problem is that it now mistakenly fires on the `if` part as described above.